src/librustdoc/html/render/print_item.rs

```diff
@@ -429,17 +446,36 @@ fn item_trait(w: &mut Buffer, cx: &Context<'_>, it: &clean::Item, t: &clean::Tra
         } else {
             // FIXME: we should be using a derived_id for the Anchors here
             w.write_str("{\n");
+            let mut toggle = false;
+
+            // If there are too many associated types, hide _everything_
+            if should_hide_fields(types.len()) {
+                toggle = true;
+                toggle_open(w, "associated items");
+            }
             for t in &types {
                 render_assoc_item(w, t, AssocItemLink::Anchor(None), ItemType::Trait, cx);
                 w.write_str(";\n");
             }
+            // If there are too many associated constants, hide everything after them
+            // We also do this if the types + consts is large because otherwise we could
+            // render a bunch of types and _then_ a bunch of consts just because both were
+            // _just_ under the limit
+            if !toggle & should_hide_fields(types.len() + consts.len()) {
+                toggle = true;
+                toggle_open(w, "associated constants and methods");
+            }
             if !types.is_empty() && !consts.is_empty() {
                 w.write_str("\n");
             }
             for t in &consts {
                 render_assoc_item(w, t, AssocItemLink::Anchor(None), ItemType::Trait, cx);
                 w.write_str(";\n");
             }
+            if !toggle & should_hide_fields(required.len() + provided.len()) {
```

Comment:
I'm wondering: is it voluntary?

Reply:
What do you mean?

Reply:
Are you asking if there's a pref? There isn't one right now, I was thinking we'd add one if people _really_ wanted one; but also: 12 is a pretty large threshold so I'm not sure if people really will.

Reply:
The single `&`. You have it above too so I'm wondering.

Reply:
Not talking about the number of items, simply the `&` instead of `&&`. With a diff it might be simpler to see:

```suggestion
            if !toggle && should_hide_fields(required.len() + provided.len()) {
```

Reply:
Ah! That's a typo, my bad!

Reply:
;)